internal/controller/controllers/infraenv_controller.go

```diff
@@ -818,11 +818,37 @@ func (r *InfraEnvReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return reply
 	}
 
+	mapPullSecretToInfraEnv := func(pullSecret client.Object) []reconcile.Request {
+		log := logutil.FromContext(context.Background(), r.Log).WithFields(
+			logrus.Fields{
+				"pull_secret":           pullSecret.GetName(),
+				"pull_secret_namespace": pullSecret.GetNamespace(),
+			})
+		infraEnvs := &aiv1beta1.InfraEnvList{}
+		if err := r.List(context.Background(), infraEnvs); err != nil {
```

Comment:
This looks expensive to compute on every change to a secret.  Can we add a reference from the secret to the InfraEnv (e.g., annotation)?

Reply:
Hi Avishay, thanks for the feedback!
So, to clarify, it won't actually reconcile unless it finds a matching infraEnv with mapPullSecretToInfraEnv().

I was just wondering: how would adding that annotation change the flow?
Say we did that; we still need to trigger a reconciliation loop per each pull secret change. How would that look

Reply:
This code that scans each InfraEnv will be executed on every change to a Secret.  Instead, maybe the InfraEnv controller could annotate the Secret, and this method would be more efficient - check if the Secret has the annotation rather than scanning InfraEnvs.

Reply:
What do we do with existing secrets?

Reply:
They would get the annotation upon the next InfraEnv reconcile.